test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -552,9 +568,66 @@ public int liveMemberCount()
         }).call();
     }
 
-    public int nodetool(String... commandAndArgs)
+    public NodeToolResult nodetoolResult(boolean withNotifications, String... commandAndArgs)
     {
-        return sync(() -> new NodeTool(new InternalNodeProbeFactory()).execute(commandAndArgs)).call();
+        return sync(() -> {
+            DTestNodeTool nodetool = new DTestNodeTool(withNotifications);
+            int rc =  nodetool.execute(commandAndArgs);
+            return new NodeToolResult(commandAndArgs, rc, new ArrayList<>(nodetool.notifications.notifications), nodetool.latestError);
+        }).call();
+    }
+
+    private static class DTestNodeTool extends NodeTool {
+        private final StorageServiceMBean storageProxy;
+        private final CollectingNotificationListener notifications = new CollectingNotificationListener();
+
+        private Throwable latestError;
+
+        DTestNodeTool(boolean withNotifications) {
+            super(new InternalNodeProbeFactory(withNotifications));
```

Comment:
since we are creating `InternalNodeProbe` instance to get `storageProxy ` in the next line. It looks like we don't need `withNotifications` for `InternalNodeProbeFactory`..

Reply:
do you mean 

```
storageProxy = new InternalNodeProbe(withNotifications).getStorageService();
```

Need it there as well.  `storageProxy` is what allows notifications so need to disable here as well.